lib/app.dart

```diff
@@ -12,26 +13,39 @@ var links = [
 ];
 
 class App {
+  app_view.View view;
+  login_controller.Controller loginController;
   nav_controller.Controller navController;
+
   App() {
+    loginController =
+        login_controller.Controller(_onSignInComplete, _onSignOutComplete);
     // TO THINK ABOUT: navController might want to take in the respective controllers as well
     navController = nav_controller.Controller(links);
-    _handlePath(null);
-    utils.listenToPathChanges(_handlePath);
+    path.listenToChanges(_handlePath);
+  }
+
+  void _onSignInComplete() async {
+    view.showLoadingIndicator();
+    logger.debug('Loading all data ...');
+    view.hideLoadingIndicator();
+  }
+
+  void _onSignOutComplete() {
+    logger.debug('Clearing all data ...');
   }
 
-  void _handlePath(html.Event event) {
-    logger.debug('updated url to ${utils.currentPathname}');
-    switch (utils.currentPathname) {
+  void _handlePath(_) {
+    logger.debug('updated url to ${path.currentName}');
+    switch (path.currentName) {
```

Comment:
The comment below in 'path.dart` might change this function, but a comment for the future: in general, it's not a good idea in a function to read and rely on state from another library.

In this case, this function should take the relative path as the only argument, and then it would be the job of the caller to do the required work to prepare the argument and pass it on (e.g. in this case, absorb the event and extract the location name).